### PR TITLE
TATB-05 : [BUG] Add login link to user registration form login form

### PR DESCRIPTION
**Pull request by PrabhatJTDeveloper**
…nent

### CODE REVIEW

File: src/app/authorization/registration/registration.component.html [hunk 1]
```diff
@@ -1,231 +1,244 @@
...
-          >
-            Number is long.
+        <div class="row">
+          <div class="col">
+            <div class="field">
+              <label for="birthday" class="form-label"> Date Of Birth :</label>
+              <input
+                type="date"
+                class="form-control input-sm"
+                formControlName="birthday"
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.html]: Can you please change controller name ?

File: src/app/authorization/registration/registration.component.scss [hunk 2]
```diff
@@ -1,33 +1,30 @@
...
+  margin-bottom: 4rem;
 }
-input[type="date"]{
-    margin-left:20px;
-    padding:3px;
-    font-family: "Roboto Mono",monospace;
-    border-radius: 4px;
-    width:320px;
-    margin-top: 10px;
+input[type="date"] {
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.scss]: Please use class instead

File: src/app/authorization/registration/registration.component.scss [hunk 3]
```diff
@@ -1,33 +1,30 @@
...
+  border-radius: 4px;
 }
-.cpwd{
-    margin-top: 10px;
+.field {
+  margin-top: 15px;
 }
-.gender{
-    margin-top: 10px;
+.form {
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.scss]: Why form class direct ?

File: src/app/authorization/registration/registration.component.scss [hunk 4]
```diff
@@ -1,33 +1,30 @@
...
-    border-radius: 4px;
-    width:320px;
-    margin-top: 10px;
+input[type="date"] {
+  font-family: "Roboto Mono", monospace;
+  border-radius: 4px;
 }
-.cpwd{
-    margin-top: 10px;
+.field {
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.scss]: Please change to diffrent class name

File: src/app/authorization/registration/registration.component.ts [hunk 5]
```diff
@@ -1,164 +1,120 @@
...
+  FormBuilder,
   FormGroup,
   ValidationErrors,
   ValidatorFn,
   Validators,
 } from '@angular/forms';
 import { Router } from '@angular/router';
 
+import { DialogService } from 'src/app/dialog-service/dialog.service';
+
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.ts]: Remove white space

File: src/app/authorization/registration/registration.component.ts [hunk 6]
```diff
@@ -1,164 +1,120 @@
...
-    fullname: '',
-    email: '',
-    pwd: '',
-    cpwd: '',
-    gender: '',
-    birthday: '',
-    mobile: '',
-  };
+  todayDate: string | undefined = new Date().toISOString().slice(0, 10);
+  registerUser: any = [];
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.ts]: please change any type as already declared as array

File: src/app/authorization/registration/registration.component.ts [hunk 7]
```diff
@@ -1,164 +1,120 @@
...
-    return this.registrationForm.get('birthday') as FormControl;
-  }
-  get Mobile(): FormControl {
-    return this.registrationForm.get('mobile') as FormControl;
+  validateNumber: ValidatorFn = (
+    control: AbstractControl
+  ): ValidationErrors | null => {
+    const regex = /^(?:(?:\+|0{0,2})91(\s*[\-]\s*)?|[0]?)?[123456789]\d{9}$/;
+    return regex.test(control.value) ? null : { pattern: true };
+  };
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.ts:85]: Please add space here

File: src/app/authorization/registration/registration.component.ts [hunk 8]
```diff
@@ -1,164 +1,120 @@
...
+      ],
+      gender: ['', Validators.compose([Validators.required])],
+      birthday: ['', Validators.compose([Validators.required])],
+      mobile: [
+        '',
+        Validators.compose([Validators.required, this.validateNumber]),
+      ],
+      acceptTerms: [false, Validators.requiredTrue],
+    });
   }
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.ts:116]: Please add space here

**Review by charmijtdev — CHANGES_REQUESTED** [review]

File: src/app/authorization/registration/registration.component.html [hunk 9]
```diff
@@ -1,237 +1,240 @@
...
+        <div class="form-group form-check">
+          <div class="form-container">
             <input
-              type="text"
-              class="form-control"
-              id="mobile"
-              formControlName="mobile"
+              type="checkbox"
+              formControlName="acceptTerms"
+              id="acceptTerms"
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.html]: what is use for id here ?

File: src/app/authorization/registration/registration.component.html [hunk 10]
```diff
@@ -1,237 +1,240 @@
...
-            </div>
-            <div
-              class="form-text text-danger"
-              *ngIf="Mobile.errors?.['maxlength'] && (Mobile.dirty || Mobile.touched)"
+            <label for="acceptTerms" class="form-check-label"
+              >Accept Terms & Conditions</label
             >
-              Number is long.
-            </div>
             <div
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.html]: I guess no  need for this pleaser remove this

File: src/app/authorization/registration/registration.component.ts [hunk 11]
```diff
@@ -1,134 +1,120 @@
...
-    this.initForm();
+    this.createForm();
+    this.getRegistredUser();
   }
-
-  public submitform() {
-    const formValue = this.registrationForm.value;
-    if (!localStorage.getItem('registeruser')) {
-      this.registeruser = [];
+  getRegistredUser() {
```
Comment (charmijtdev) [inline src/app/authorization/registration/registration.component.ts:38]: please add space here and remove condition for below to match

**Review by charmijtdev — CHANGES_REQUESTED** [review]

**Review by charmijtdev — APPROVED** [review]

**Review by PoojanPatelJT — APPROVED** [review]

**Pull request closed without merging**